draft-ietf-mls-protocol.md

```diff
@@ -2366,13 +2365,11 @@ chain from the committer's leaf to the root.
 
 ## Update Paths
 
-As described in {{commit}}, each MLS Commit message may optionally
-transmit a LeafNode and parent node values along its direct path.
-The path contains a public key and encrypted secret value for all
-intermediate nodes in the filtered direct path from the leaf to the
-root. The path is ordered
-from the closest node to the leaf to the root; each node MUST be the
-parent of its predecessor.
+As described in {{commit}}, each Commit message may optionally contain an
+UpdatePath, with a new LeafNode and set of parent nodes for the sender's
+filtered direct path. For each intermediate node, the UpdatePath contains a new
+public key and encrypted path secret. The nodes are ordered such that each node
+is the ancestor of its predecessor.
```

Comment:
```suggestion
is the next non-blank ancestor of its predecessor in the filtered direct path.
```

Reply:
I don't think this is right, the filtered direct path doesn't have blank nodes.

Reply:
I can see how it's confusing to say both.  How about either "next non-blank ancestor" or "next ancestor in the filtered direct path"?

Alternatively, the filtered direct path already has an order defined:

> The direct path of a root is the empty list, and of any other node is the concatenation of that node's parent along with the parent's direct path.

> The filtered direct path of a node is obtained from the node's direct path by removing all nodes ... [note this doesn't change the order]

So we could just say that the nodes in the UpdatePath correspond to the nodes in the filtered direct path.